src/parallel/global_memory/g_common.cu

```diff
@@ -0,0 +1,67 @@
+#include "../common/memory_management.cuh"
+#include "../common/statistics.h"
+#include "../kernels/iteration_step.cuh"
+
+
+int main(int argc, char *argv[])
+{
+    CA *d_ca = new CA();
+    CA *h_ca = new CA();
+    double *headsWrite;
+
+    h_ca->heads = new double[ROWS * COLS]();
+    h_ca->Sy = new double[ROWS * COLS]();
+    h_ca->K = new double[ROWS * COLS]();
+    h_ca->sources = new double[ROWS * COLS]();
+
+    initializeCA(h_ca);
+
+    allocateMemory(d_ca, headsWrite);
+    copyDataFromCpuToGpu(h_ca, d_ca);
+
+    dim3 blockSize(BLOCK_SIZE, BLOCK_SIZE);
+    const int blockCount = ceil((double) (ROWS * COLS) / (BLOCK_SIZE * BLOCK_SIZE));
+    int gridSize = ceil(sqrt(blockCount));
+    dim3 gridDims(gridSize, gridSize);
+
+    Timer stepTimer{};
+    startTimer(&stepTimer);
+
+    for (unsigned i{}; i < SIMULATION_ITERATIONS; ++i)
+    {
+#ifdef STANDARD
+        kernels::standard_step <<< gridDims, blockSize >>>(*d_ca, headsWrite);
+#endif
+#ifdef HYBRID
+        kernels::hybrid_step <<< gridDims, blockSize >>>(*d_ca, headsWrite);
+#endif
+#ifdef SHARED
+        kernels::shared_step <<< gridDims, blockSize >>>(*d_ca, headsWrite, gridSize);
```

Comment:
Może usuń już to przekazywanie gridSize'u i użyj gridDim w kernelu, analogicznie do tego co mi zaproponowałeś w w active cellach.

Reply:
done